src/hooks/useOnOutsideClick.tsx

```diff
@@ -0,0 +1,59 @@
+import React, {useEffect, useCallback} from 'react'
+
+export type TouchOrMouseEvent = MouseEvent | TouchEvent
+
+export type UseOnOutsideClickProps = {
```

Comment:
Can I suggest that we reserve the term `Props` for React component props? For `useFocusTrap` I called the options interface `FocusTrapHookSettings`.

Reply:
Sure!